cmd/install/assets/hypershift_operator.go

```diff
@@ -477,10 +490,28 @@ func (o HyperShiftServiceMonitor) Build() *prometheusoperatorv1.ServiceMonitor {
 				{
 					Interval: "30s",
 					Port:     "metrics",
+					RelabelConfigs: []*prometheusoperatorv1.RelabelConfig{
```

Comment:
I'm not sure I follow. Once I fixed RBAC locally and fixed up the labels on the hypershift operator deployment as an experiment (adding `app=hypershift-operator` and `hypershift.openshift.io/control-plane-component=hypershift-operator`) then the following query does what we want:

```
sum by (app, namespace) (
  sum(rest_client_requests_total) by (pod, namespace)
* on (pod) group_left(app)
  label_replace(kube_pod_labels{label_hypershift_openshift_io_control_plane_component!=""}, "app", "$1", "label_app", "(.*)")
)
```

output:

```
{app="hypershift-operator", namespace="hypershift"} | 368
```

Of course we might want separate recording rules for hypershift itself versus hosted control planes, but same thing applies. As the query shows, there's no need for any special labeling rules on the monitors to isolate the metrics to pods matching our labels and attaching the pod app label to the requests metric.

Reply:
> then the following query does what we want:

Sure, updated it to use the join. But at the end of the day the result of the two is identical, but using the relabel_config has the advantage that:
* We don't depend on kube-state-metrics to provide us with the label information
* We don't need to do an compuationally expensive and hard to understand join

> We could have two different rule groups to the PrometheusRule

The only thing the grouping does to my knowledge is display it together or not in the Prometheus UI. I think it would be good to keep all hypershift-related rules in one group, at least until we have a two-digit number of rules.